src/d3a/models/strategy/external_strategy.py

```diff
@@ -75,14 +75,29 @@ def _accept_offer_response_channel(self):
     def _list_offers_response_channel(self):
         return f"{self._list_offers_channel}/response"
 
+    @property
+    def _bid_response_channel(self):
+        return f"{self._bid_channel}/response"
+
+    @property
+    def _delete_bid_response_channel(self):
+        return f"{self._delete_bid_channel}/response"
+
     def sub_to_external_requests(self):
-        self.pubsub.subscribe(**{
-            self._offer_channel: self._offer,
-            self._delete_offer_channel: self._delete_offer,
-            self._accept_offer_channel: self._accept_offer,
-            self._list_offers_channel: self._offer_lists
-        })
-        self.pubsub.run_in_thread(daemon=True)
+        if ConstSettings.IAASettings.MARKET_TYPE == 1:
+            self.redis_db.sub_to_multiple_channels({
+                self._offer_channel: self._offer,
+                self._delete_offer_channel: self._delete_offer,
+                self._accept_offer_channel: self._accept_offer,
+                self._list_offers_channel: self._offer_lists,
+            })
+        else:
+            self.redis_db.sub_to_multiple_channels({
+                self._offer_channel: self._offer,
+                self._delete_offer_channel: self._delete_offer,
+                self._bid_channel: self._bid,
+                self._delete_bid_channel: self._delete_bid
```

Comment:
Correct me if Im wrong but shouldn't there be one more channel for `accepting bid`

Reply:
Like my comment to Hannes above, the external connection should not be able to accept bids for 2-sided markets, since the market is responsible for the market clearing (accepting bids and offers). The external connection is only responsible for placing offers and bids to a market, the market itself performs the trade.

Reply:
ah okay makes sense, thanks for explaining.